metrics/metrics.go

```
@@ -53,63 +53,69 @@ func Flush() error {
 }
 
 // Gauge measures the value of a metric at a particular time.
-func Gauge(name string, value float64, tags ...string) {
-	err := statsdClient.Gauge(name, value, tags, opts.metricSampleRate)
+func Gauge(name string, value float64, options ...MetricOpts) {
```

With this impl, do we need to return an error? Wdyt @bendiknesbo

---

If we return error, then we **allow** ourselves to add validation at a later point, if we want to do that. This can be useful if we want to add other types of `MetricOpts` later (e.g. for new features in DogStatsD, or if we want to migrate everything to OpenTelemetry or similar).

The signature can still be without returning an error, and if there is an error, then we can call `opts.errorHandler` and return. This keeps the signature clean (no need to deal with errors everywhere you use metrics), but still allow for flexibility for handling errors on a case-by-case basis.